### PR TITLE
Documentation update

### LINKED ISSUE #137: Document that Destructure config does not work in current Nuget release
Currently, the dev branch is the default branch for this repo. When I visit the project URL listed under the currently released nuget package, I see the readme for the dev branch, which documents features (like Destructure configuration) that aren't actually present in the currently released version of serilog.settings.configuration. 

Would it be possible to have a note on the readme in dev to give a hint that certain features are only available in pre-release packages?

### PR DESCRIPTION
**Pull request by skomis-mm**
* Fixes #137, #152, #154.
* improvements for making user contribution story (grab, build / test / run) easier under Linux environment. Added VSCode config that points to Sample app.

[Readme.md](https://github.com/serilog/serilog-settings-configuration/blob/f6d60233003a883fbdc9ed3268ed0f9459a7e587/README.md) was restructured to:

1. Basic setup / config
2. Syntax description and features 
3. Arguments binding
4. Samples

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,7 +1,85 @@
...
+* #161 - support simple type names for Serilog types
+* #151 - no longer rely on static state in ConfigurationReader
+* #179 - added missing null checks for settingConfiguration
+* #163 - added new ReadFrom.Configuration(...) overloads; marked old as obsolete
+* #176 - added test to show how to filter child contexts
+
+3.0.1
+
+* #142 - Fix IConfiguration parameters not being populated
+* #143 - Fix ReadFrom.ConfigurationSection() looking for sections below a root Serilogsection
```
Comment (sungam3r) [inline CHANGES.md:27]:
```suggestion
* #143 - Fix ReadFrom.ConfigurationSection() looking for sections below a root Serilog section
```

Comment (skomis-mm) [reply CHANGES.md]: fixed, thanks

**Comment by tsimbalar** [comment]
Wow, great work to make newcomers life easiers, and really neat job on completing the documentation 👍 

I'm now wondering if https://github.com/tsimbalar/serilog-settings-comparison still makes sens 🤔  (or at least I should try to update it with new supported features)

**Comment by skomis-mm** [comment]
@tsimbalar  I think it makes sense to update `serilog-settings-comparison`. Documentation here is about syntax and tied to `Microsoft.Settings.Configuration` , but yours more focused on differences and how one configuration type can be replaced with other and what are the limits.

**Pull request merged**